src/script/api/script_event_types.hpp

```diff
@@ -1060,4 +1060,44 @@ class ScriptEventRoadReconstruction : public ScriptEventCompanyTown {
 	static ScriptEventRoadReconstruction *Convert(ScriptEventCompanyTown *instance) { return (ScriptEventRoadReconstruction *)instance; }
 };
 
+/**
+ * Event VehicleAutoReplaced, indicating a vehicle has been auto replaced.
+ * @api ai
+ */
+class ScriptEventVehicleAutoReplaced : public ScriptEvent {
+public:
+	/**
+	 * @param old_id The vehicle that has been replaced.
+	 * @param new_id The vehicle that has been created in replacement.
+	 */
+	ScriptEventVehicleAutoReplaced(VehicleID old_id, VehicleID new_id) :
+		ScriptEvent(ET_VEHICLE_AUTOREPLACED),
+		old_id(old_id),
+		new_id(new_id)
+	{}
+
+	/**
+	 * Convert an ScriptEvent to the real instance.
+	 * @param instance The instance to convert.
+	 * @return The converted instance.
+	 */
+	static ScriptEventVehicleAutoReplaced *Convert(ScriptEvent *instance) { return (ScriptEventVehicleAutoReplaced *)instance; }
+
+	/**
+	 * Get the VehicleID of the vehicle that has been replaced.
+	 * @return The VehicleID of the vehicle that has been replaced.
```

Comment:
Should perhaps detail that the VehicleID is no longer valid. Perhaps: "The VehicleID of the vehicle before replacement. This ID is no longer valid for referencing the vehicle."